### PR TITLE
Create BEIR benchmark doc

### LINKED ISSUE #26: Create BEIR benchmark doc
Create BEIR benchmark doc

### PR DESCRIPTION
**Pull request by minsing-jin**
close #26

### CODE REVIEW

File: ragchain-structure/benchmark/dataset-evaluator/beir.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+remove_path(bm25_path, pickle_path)
+```
+As you can see, `generate_path` function generate bm25 path and pikle path
+matched dataset name. And, `set_beir_evaluator` function generate
+`retrieval`, `db`, `llm`, `pipeline`. If you want to modify them, you can
+set others in `set_beir_evaluator`.
+
+# Example Use
+NOTICE:
+We except context_recall metric that is ragas metric.
```
Comment (vkehfdl1) [inline ragchain-structure/benchmark/dataset-evaluator/beir.md]: We do not support context_recall metric, which is from `ragas`.

File: ragchain-structure/benchmark/dataset-evaluator/beir.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+```Python
+from RAGchain.benchmark.dataset import <your beir evaluator>
+
+pipeline = <your pipeline>
+retrievals = [<your retrieval>]
+db = <your db>
+
+evaluator = <your beir evaluator>(pipeline, evaluate_size=5)
+evaluator.ingest(retrievals=[retrieval], db=db, ingest_size=20) # ingest dataset to db and retrievals
```
Comment (vkehfdl1) [inline ragchain-structure/benchmark/dataset-evaluator/beir.md]: [retrieval] => retrievals

File: ragchain-structure/benchmark/dataset-evaluator/beir.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+Beir test code is little different from other benchmark.
+```Python
+bm25_path, pickle_path = generate_path('fever')
+retrieval, db, llm, pipeline = set_beir_evaluator(bm25_path, pickle_path)
+evaluator = <Beir evaluator>(run_pipeline=pipeline, evaluate_size=5)
+evaluator.ingest(retrievals=[retrieval], db=db, ingest_size=20)
+
+yield evaluator
+remove_path(bm25_path, pickle_path)
```
Comment (vkehfdl1) [inline ragchain-structure/benchmark/dataset-evaluator/beir.md]: delete this code
No need for tutorial, and only need for pytest fixture;;

File: ragchain-structure/benchmark/dataset-evaluator/beir.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+![beir-structure.png](../../../.gitbook/assets/beir-structure.png)
+
+Also, we removed non-relevant score passages at qrels.<br>
+The significance of this dataset is diverse tasks, diverse domains, task difficulties, and diverse annotation strategies
+in each dataset.
+
+Beir test code is little different from other benchmark.
+```Python
+bm25_path, pickle_path = generate_path('fever')
+retrieval, db, llm, pipeline = set_beir_evaluator(bm25_path, pickle_path)
```
Comment (vkehfdl1) [inline ragchain-structure/benchmark/dataset-evaluator/beir.md]: also, where is set_beir_evaluator?
Don't just copy and paste pytest fixture codes. Write adequate code for newbies.

File: ragchain-structure/benchmark/dataset-evaluator/beir.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+Basic preprocessing process is same at 6 benchmark dataset,
+So we construct preprocess process, ingest, evaluation code are in BaseBeirEvaluator class.
+Each evaluator class corresponding to the benchmark datasets
+(FEVER, FIQA, HOTPOTQA, QUORA, SCIDOCS, and SCIFACT) inherits from the BaseBeirEvaluator.
+
+![beir-structure.png](../../../.gitbook/assets/beir-structure.png)
+
+Also, we removed non-relevant score passages at qrels.<br>
+The significance of this dataset is diverse tasks, diverse domains, task difficulties, and diverse annotation strategies
+in each dataset.
```
Comment (vkehfdl1) [inline ragchain-structure/benchmark/dataset-evaluator/beir.md]: useless annotation

File: ragchain-structure/benchmark/dataset-evaluator/beir.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+evaluator = <Beir evaluator>(run_pipeline=pipeline, evaluate_size=5)
+evaluator.ingest(retrievals=[retrieval], db=db, ingest_size=20)
+
+yield evaluator
+remove_path(bm25_path, pickle_path)
+```
+As you can see, `generate_path` function generate bm25 path and pikle path
+matched dataset name. And, `set_beir_evaluator` function generate
+`retrieval`, `db`, `llm`, `pipeline`. If you want to modify them, you can
+set others in `set_beir_evaluator`.
```
Comment (vkehfdl1) [inline ragchain-structure/benchmark/dataset-evaluator/beir.md]: You didn't make generate_path and set_beir_evaluator function for BeirEvaluator. Usage of BeirEvaluator is literally same with other evaluator, right?
You dind't read this AI generated documentation, right?

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]

**Comment by minsing-jin** [comment]
All pull requests about benchmark docs are unified in issue #24.

**Pull request closed without merging**